### PR TITLE
Max value calculation wrong due to cutoff when max value is not a multiple of step

### PR DESCRIPTION
**Pull request by thaitzer**
gracefully handle max value when a step size is defined and the max v…alue does not fit the step size

### CODE REVIEW

**Comment by seiyria** [comment]
please add a unit test so this particular behavior doesn't happen anymore!

**Comment by seiyria** [comment]
additionally, please make sure this passes our linter!

**Comment by thaitzer** [comment]
I  have added to test and checked the linter.

However I have a problem with my test code. I did not find a useful way to get the outer right position for the slider right - i had to add 10px manually to the mouse-click. I do not know why this is the case but I figure you guys probably have a pointer for me :-). Then I will gladly fix the test and get rid of the "+10".

For the mouse-clicking I copied the method from another test - this is where codeclimate complains now about code duplication. What is your suggested way of fixing this?

File: test/specs/StepReachMaxValueSpec.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+	var slider;
+	var options;
+
+	describe('max value should be reached', function() {
+		beforeEach(function() {
+			options = {
+				min: 39,
+				max: 1310,
+				step: 5,
+        scale:"logarithmic",
```
Comment (rovolution) [inline test/specs/StepReachMaxValueSpec.js]: test is solid! could you fix the spacing on this line?

File: test/specs/StepReachMaxValueSpec.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+				value: 44
+			};
+
+			slider = new Slider(document.getElementById(SLIDER_ID), options);
+		});
+
+		it("Value should contain max value when slider is moved to outer right position", function() {
+
+			var sliderLeft = slider.sliderElem.offsetLeft;
+      var offsetY = slider.sliderElem.offsetTop;
```
Comment (rovolution) [inline test/specs/StepReachMaxValueSpec.js]: spacing

File: test/specs/StepReachMaxValueSpec.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+			var offsetX = sliderLeft + slider.sliderElem.clientWidth+10;
+
+      var expectedValue = slider.options.max;
```
Comment (rovolution) [inline test/specs/StepReachMaxValueSpec.js]: spacing

**Comment by rovolution** [comment]
> However I have a problem with my test code. I did not find a useful way to get the outer right position for the slider right - i had to add 10px manually to the mouse-click. I do not know why this is the case but I figure you guys probably have a pointer for me :-). Then I will gladly fix the test and get rid of the "+10".

as long as you feel the +10 value is roughly accurate, that is fine. maybe include a comment in the test about why you are using this value

> For the mouse-clicking I copied the method from another test - this is where codeclimate complains now about code duplication. What is your suggested way of fixing this?

Don't worry about this. I have been pondering completely removing the codeclimate check or just limiting it to the source directory.

**Comment by thaitzer** [comment]
I clean up the test code, added the comment and fixed the intendation.

**Comment by rovolution** [comment]
@thaitzer just pulled down your branch to run the examples page and it works great!

Thanks again for your contribution!

**Comment by rovolution** [comment]
@thaitzer btw, are there any existing issues from our issues list that this fix addresses?

I just want to ensure those get listed in our changelog.

https://github.com/seiyria/bootstrap-slider/issues

**Comment by rovolution** [comment]
@thaitzer published to v9.8.1: https://github.com/seiyria/bootstrap-slider/blob/master/CHANGELOG.md#981--2017-07-06

**Comment by thaitzer** [comment]
@rovolution I did not check the issues before writing this mini-patch. So the answer is: not that I know of.

While I am writing this: I never bothered to check if the same thing happens on the linear scale - I might be a good idea to do that.

**Comment by rovolution** [comment]
> While I am writing this: I never bothered to check if the same thing happens on the linear scale - I might be a good idea to do that.

if you find that this is also an issue, we would definitely appreciate another PR!

**Pull request merged**